File: 2019W38/preppindataw38.R
library(dplyr)
library(readr)

visit <- read_csv("E:/Input Data.csv", 
                  col_types = cols(`Date of Servce` = col_date(format = "%m/%d/%Y"))) %>%
  group_by(`PatientID`) %>%
  arrange(`Date of Servce`) %>%
  mutate('Patient Visit Number' = cumsum(`PatientID`/`PatientID`),
         'First Visit Date' = min(`Date of Servce`),
         'Total Patient Visits' = max(`Patient Visit Number`),
         'New Patient Flag' = ifelse(`Patient Visit Number`==1,'New Patient','Returning Patient'))
  
View(visit)

#check <- visit %>% filter(`PatientID`==94)

File: 2019/2019W38/PreppindataW38.py
import pandas as pd

visit = pd.read_csv("E:\Input Data.csv")
visit['Date of Servce'] = visit['Date of Servce'].apply(lambda x: pd.to_datetime(x).date())
visit['Patient Visit Number'] = visit.sort_values('Date of Servce').groupby('PatientID').cumcount()+1
visit['First Visit Date'] = visit['Date of Servce'].groupby(visit['PatientID']).transform('min')
visit['Total Patient Visits'] = visit['Patient Visit Number'].groupby(visit['PatientID']).transform('max')
visit['New Patient Flag'] =visit['Patient Visit Number'].apply(lambda x: 'New Patient' if x==1 else 'Returning Patient')

#check = visit[visit['PatientID']==94]
